compose-app/src/main/java/com/mapbox/maps/extension/compose/ComposeUtils.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.maps.extension.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import com.mapbox.maps.MapSurface
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettings
+import com.mapbox.maps.plugin.gestures.generated.GesturesSettingsInterface
+import kotlinx.coroutines.awaitCancellation
+
+
+internal suspend inline fun disposingComposition(factory: () -> Composition) {
+  val composition = factory()
+  try {
+    awaitCancellation()
+  } finally {
+    composition.dispose()
+  }
+}
+
+internal fun MapView.newComposition(
+  parent: CompositionContext,
+  content: @Composable () -> Unit
+): Composition {
+  return Composition(
+    MapApplier(this, this), parent
+  ).apply {
+    setContent(content)
+  }
+}
+
+internal fun MapSurface.newComposition(
+  parent: CompositionContext,
+  content: @Composable () -> Unit
+): Composition {
+  return Composition(
+    MapApplier(this, this), parent
+  ).apply {
+    setContent(content)
+  }
+}
+
+
+public fun GesturesSettingsInterface.applySettings(gesturesSettings: GesturesSettings) {
```

Comment:
just curious, could this function be inside sdk?

Reply:
for v11 yes